### PR TITLE
Improve texture creation validation

### PR DESCRIPTION
**Pull request by kvark**
**Connections**
Fixes https://github.com/gfx-rs/wgpu-rs/issues/528#issuecomment-678853886

**Description**
Improves the mipmap levels check.
Note that it also removes the `TooManyLayers` error, and instead produces max `u16` value. This is to be followed up with a check for *real* limits, which has to happen regardless, and trying to catch it on the number conversion step seems wasteful.

**Testing**

### CODE REVIEW

File: wgpu-core/src/resource.rs [hunk 1]
```diff
@@ -215,8 +215,8 @@ pub struct Texture<B: hal::Backend> {
 
 #[derive(Clone, Debug, Error)]
 pub enum TextureDimensionError {
-    #[error("too many layers ({0}) for texture array")]
-    TooManyLayers(u32),
+    #[error("one of the dimensions is zero")]
+    Zero,
```
Comment (cwfitzgerald) [inline wgpu-core/src/resource.rs:219]: Could use [`TextureErrorDimension`](https://github.com/gfx-rs/wgpu/blob/master/wgpu-core/src/command/transfer.rs#L36) to make this error better.

Comment (cwfitzgerald) [reply wgpu-core/src/resource.rs:219]: And if you want to get real fancy, to detect this, you could do

```rust
match (width, height, depth) {
    (0, _, _) => return Err(Zero(TextureErrorDimension::X)),
    (_, 0, _) => return Err(Zero(TextureErrorDimension::Y)),
    (_, _, 0) => return Err(Zero(TextureErrorDimension::Z)),
    _ => {}
}
```

**Comment by kvark** [comment]
Thanks! Using that enum now.
bors r=cwfitzgerald

**Comment by kvark** [comment]
bors r=cwfitzgerald

**Pull request merged**